include/repeater_cadmium.hpp

```diff
@@ -0,0 +1,197 @@
+*/
+
+#ifndef __REPEATER_CADMIUM_H__
+#define __REPEATER_CADMIUM_H__
+
+#include <cadmium/modeling/ports.hpp>
+#include <cadmium/modeling/message_bag.hpp>
+#include <limits>
+#include <math.h>
+#include <assert.h>
+#include <memory>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
+#include <algorithm>
+#include <limits>
+#include <random>
+
+#include "message.hpp"
+
+using namespace cadmium;
+using namespace std;
+/**
+ * Structure that holds input and output messages
+ * using input and output port.
+*/
+struct repeater_defs {
+    struct packet_sent_out : public out_port<Message_t> {
+    };
+    struct ack_received_out : public out_port<Message_t> {
+    };
+    struct packet_in : public in_port<Message_t> {
+    };
+    struct ack_in : public in_port<Message_t> {
+    };
+};
+
+/**
+ * The Sender class sends out messages and receives acknowledges.
+*/
+template<typename TIME>
+class Repeater {
+    // putting definitions in context
+    using defs = repeater_defs;
+public:
+    TIME PREPARATION_TIME;    /**< Constant that holds the time delay from acknowledge to output. */ //!<Time delay constant.
+
+    /**
+	 * Constructor for Repeater class.
+	 * Initializes the delay constants and state structure.
+	*/
+    Repeater() noexcept {
+        PREPARATION_TIME = TIME("00:00:10");
+        state.ack    = 0;
+        state.packet = 0;
+    }
+
+    /**
+      * Structure that holds the state variables for packet
+      * acknowledgement packet and to check if sending packet
+      * or acknowledgement.
+    */
+    struct state_type {
+        bool ack;
+        bool sending;
+        int packet;
+        int ack_packet;
+    };
+    state_type state;
+
+    /**
+     * Defining input port and output port
+    */
+    using input_ports = std::tuple<typename defs::packet_in,
+        typename defs::ack_in>;
+    using output_ports = std::tuple<typename defs::packet_sent_out,
+        typename defs::ack_received_out>;
+
+    /**
+     * Internal transition function that sets the
+     * state of sending packet and acknowledgement
+     * to passive.
+    */
+    void internal_transition() {
+        state.sending = false;
+        state.ack = false;
+    }
+
+    /**
+     * Function that performs external transition.
+     * Retrieves the messages: if the number of messages
+     * is more than 1, it asserts that only one message is
+     * expected per time unit. After that it determines
+     * from which input port message is coming and based
+     * on the message type sets the state.
+     * @param e time variable
+     * @param mbs message bags
+    */
+    void external_transition(TIME e,
+                             typename make_message_bags<input_ports>::type mbs) {
+        if ((get_messages<typename defs::packet_in>(mbs).size()
+            +get_messages<typename defs::ack_in>(mbs).size()) > 1) {
+            assert(false && "One message at a time");
+        }
```

Comment:
Same here. You need to check it twice: "Once message per connection"

Reply:
This has been fixed.